src/ScheduleFunctions.cpp

```diff
@@ -29,6 +31,31 @@ using std::tuple;
 using std::vector;
 
 namespace {
+std::ostream& operator<<(std::ostream& out, const std::vector<Function>& v) {
+    out << "{ ";
+    for (size_t i = 0; i < v.size(); ++i) {
+        out << v[i].name();
+        if (i != v.size() - 1) {
+            out << ", ";
+        }
+    }
+    out << " }";
+    return out;
+}
+
+Stmt operator>>(const Stmt& s1, const Stmt& s2)
+{
```

Comment:
nit: opening brace should be on previous line.

non-nit: using the >> operator to concatenate Stmts is... weird. I'd have a hard time figuring out the code if I saw the construct without seeing this declaration first. I'd like to see a consensus among contributors that this is better than a simple `concat()` method before it lands.

Reply:
Okay, this is easy enough to inline. It helped me make sense of things (it matches Haskell's monadic "and then") while writing it.